magnet_derive/src/codegen_field.rs

```diff
@@ -90,15 +98,17 @@ fn field_def(field: &Field) -> Result<TokenStream> {
     let max_excl = magnet_meta_name_value(&field.attrs, "max_excl")?;
     let lower = bounds_from_meta(min_incl, min_excl)?;
     let upper = bounds_from_meta(max_incl, max_excl)?;
+    let doc = doc_meta(&field.attrs).and_then(|doc| meta_value_as_str(&doc).ok()).unwrap_or_else(String::new);
 
     Ok(quote! {
-        ::magnet_schema::support::extend_schema_with_bounds(
-            <#ty as ::magnet_schema::BsonSchema>::bson_schema(),
-            ::magnet_schema::support::Bounds {
-                lower: #lower,
-                upper: #upper,
-            },
-        )
+        ::magnet_schema::support::extend_schema_with_doc(
```

Comment:
Instead of wrapping each kind of extension into its own function, and increasing nesting, let's create a single extension context object and an `extend_schema()` function. The extension context object could contain all the potential extensions as optionals, and `extend_schema()` could apply them in one go. This sounds to me like a more principled and easier-to-understand approach.